### ISSUE TITLE
lookup git.devever.net: no such host

### ISSUE THREAD

**Issue by skinkie** [hlandau/acmetool#346]
When running make:

go get git.devever.net/hlandau/acmetool/...: unrecognized import path "git.devever.net/hlandau/acmetool": https fetch: Get "https://git.devever.net/hlandau/acmetool?go-get=1": dial tcp: lookup git.devever.net: no such host

**Comment by rmamba** [comment]
I see the same issue, trying to create docker container

**Comment by masak2009** [comment]
Hi there,
repository is still missing!!!

Debian 11:
```
make
	[RELOCATE]	  
	[GO-GET]	  git.devever.net/hlandau/acmetool
go get git.devever.net/hlandau/acmetool/...: unrecognized import path "git.devever.net/hlandau/acmetool": https fetch: Get "https://git.devever.net/hlandau/acmetool?go-get=1": dial tcp: lookup git.devever.net on 213.186.33.99:53: no such host
make: *** [Makefile:76: .gotten] Chyba 1
```

```
ping git.devever.net
ping: git.devever.net: Not found
```
This problem exists for years. I can not compile acmetool on several distro of Debian (7,8,9,10,11) when git.devever.net is not exist. Owner of domain/server has to allow access to this repository or source of acmetool has to be changed. I am not go lang programmer, so i dont know what to change....

Go Lang version was 1.11 or 1.14 or 1.17

**Comment by mnalis** [comment]
Is there something wrong with [Debian-provided acmetool package](https://packages.debian.org/search?keywords=acmetool)? Why do you need to compile it? 

- To see how Debian compiles it, you can look at debian source packages like https://packages.debian.org/sid/acmetool, e.g. if you're on Bullseye:

```
apt-get install --install-recommends devscripts
apt-get build-dep acmetool
dget -x http://deb.debian.org/debian/pool/main/a/acmetool/acmetool_0.2.2-1.dsc
cd acme*/
debuild
```

- Also see related: https://github.com/hlandau/acmetool/issues/325 and linked https://github.com/hlandau/acmetool/pull/326

**Comment by masak2009** [comment]
Problem is Let's Encrypt :-) I started compiling from sources because "latest working code" exists several years only on github after LE shutdown V1 API. There is mess in versions of acmetool vs binary ready compiled versions of acmetool. I dont understand why someone building code which depends on non-existing sources as open-source. Is somewhere confirmation about which binary packages of acmetool works with API V2? FYI: I used compiling of acmetool maybe for year and it works like a charm. But later dependency source-code went away to offline and now it's impossible to build. I have/build/operate more than 500 project which uses LE & acmetool and it's impossible to wait for a months when LE change something and changes will be available here only as source code. That's why i need ready-to-compile source code.

**Comment by mnalis** [comment]
> Problem is Let's Encrypt :-) I started compiling from sources because "latest working code" exists several years only on github after LE shutdown V1 API

Since Debian Bullseye, version included there (acmetool 0.2.1) is using V2 API. So if you're on Debian Stable or newer (Testing, Sid), just do `apt-get install acmetool` and enjoy the ride. Debian "main" repository guarantees all the things are compiled (and correspond to) from sources as present in debian source pool (which you can retrieve with `apt-get source acmetool`).

> Is somewhere confirmation about which binary packages of acmetool works with API V2?

Yes.
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=954189
- https://github.com/letsencrypt/website/pull/861

> I dont understand why someone building code which depends on non-existing sources as open-source. 

It doesn't. Go compilation is (IMHO) finicky and not for the faint of heart. typing `make` won't do what you expect (compile the source you have in current directory) but instead attempt to overwrite anything you have with new thingies from the web. (or at least it seems to total Go newbie like myself).

If you follow instructions I've posted for Debian [above](https://github.com/hlandau/acmetool/issues/346#issuecomment-1523439167) it will use its own independent copy of all required acmetool sources and compile them. How to update to newer versions is beyond the scope of it, if you're interested, search for `uupdate` and `gbp` Debian tools, and see what `git clone https://salsa.debian.org/go-team/packages/acmetool.git` has. 

Unless inclined to learn a thing or two about Go, if new version appears in this github repo and contain useful things, I'd rather suggest reporting requests for newer upstream versions (or critical patches if upstream is dead) at https://bugs.debian.org/acmetool and wait for Debian maintainers to do it.

> That's why i need ready-to-compile source code.

Hopefully my instructions above helps about that, if you prefer do do your own builds instead of using Debian ones. On the other hand, this project seems to be on autopilot for years without much things happening. If LE decides to change protocol again, it might well be that you'll have to learn to program in Go (as well as setup its development environment so it doesn't try to contact no longer existing hosts which seems to be one of the problems) and implement changes yourself. So, caveat emptor.

**Comment by masak2009** [comment]
Thanks for comment. Good to know that Debian11 has acmetool for API V2.

Just for record about compiling:
I was used manual from somewhere here:
```
cd acmetool_project_from_github
make
make install
```
As I know (for all compiled things on Linux) you have to run make tu be sure that all dependencies should be load and compile to destination will work. The 2nd step is make install of course... Then you have working compiled thing in right destination of dir in Linux.

So in short: https://salsa.debian.org/go-team/packages/acmetool.git include all here missing dependencies?

Just of the record :-)
It's always funny that someone on github have to write sometning about "learning programming language XY" with advice "repair by yourself". But this is really nonsense. I have different pro skills in another programming languages and some basic skills how to sys-admin Linux distros. I work on profesional level and day has only 24HRS. No one can learn everything because of missing time. I really like to help people on github (or directly by me in real world on computer of someone) with skills that i have. These people than have time to help another people with their skills. It save time both sides in long term.

**Comment by mnalis** [comment]
> So in short: https://salsa.debian.org/go-team/packages/acmetool.git include all here missing dependencies?

Yes, packages published in Debian `main` **always** come with **full corresponding sources**, as well as with computer instructions for building the package from source (in `debian/rules` script).

> It's always funny that someone on github have to write sometning about "learning programming language XY" with advice "repair by yourself".

That was not my intention, in fact I'm in _exactly_ the same boat as you. 

It was just a friendly advice to prepare for eventualities - when some project seems to _not be actively maintained_ (like this one unfortunately looks to me at the moment), it is quite likely that _eventually it will stop working_, and one should best _plan for such contingency_ (either by switching to some other tool, or by forking the project and fixing it themselves, or by paying someone else to fix it for them, or by incentivizing original maintainer to reactivate their engagement). 

There is also _"close your eyes and hope for the best"_ method, but unfortunately it seems to only work **_sometimes_**  :smiley: , so other methods mention above might be preferred (depending on own risk assessment strategy, of course)

> I really like to help people on github (or directly by me in real world on computer of someone) with skills that i have

Me too. I have no idea about a **"go"** programming at all, but I dabble in Debian packaging, so I've tried to give you help from that side.
